### PR TITLE
[Docs] Refactor the structure of documentation

### PR DESCRIPTION
**Pull request by MengzhangLI**
Refactor the structure of documentation

### CODE REVIEW

**Comment by Junjun2016** [comment]
image
Should check the readthedoc if you update the doc.

**Comment by Junjun2016** [comment]
Could also provide the related PR.

File: .gitignore [hunk 1]
```diff
@@ -64,7 +64,7 @@ instance/
 .scrapy
 
 # Sphinx documentation
-docs/_build/
+docs/en/_build/
```
Comment (zhouzaida) [inline .gitignore:67]: We should also add docs/en/zh_cn/_build/

Comment (gaotongxiao) [reply .gitignore:67]: docs/zh_cn/_build/

File: docs/zh_cn/train.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ GPUS=16 ./tools/slurm_train.sh dev pspr50 configs/pspnet/pspnet_r50-d8_512x1024_
 您可以查看 [slurm_train.sh](../tools/slurm_train.sh) 以熟悉全部的参数与环境变量。
 
 如果您多个机器已经有以太网连接， 您可以参考 PyTorch
-[launch utility](https://pytorch.org/docs/stable/distributed_deprecated.html#launch-utility) 。
+[launch utility](https://pytorch.org/docs/en/stable/distributed_deprecated.html#launch-utility) 。
```
Comment (gaotongxiao) [inline docs/zh_cn/train.md:62]: This is a link to pytorch.org :)

File: docs/zh_cn/tutorials/customize_models.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ optimizer = dict(type='MyOptimizer', a=a_value, b=b_value, c=c_value)
 optimizer = dict(type='Adam', lr=0.0003, weight_decay=0.0001)
 ```
 
-使用者可以直接按照 PyTorch [文档教程](https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim) 去设置参数。
+使用者可以直接按照 PyTorch [文档教程](https://pytorch.org/docs/en/stable/optim.html?highlight=optim#module-torch.optim) 去设置参数。
```
Comment (gaotongxiao) [inline docs/zh_cn/tutorials/customize_models.md:45]: This is a link to pytorch.org :)

File: docs/zh_cn/tutorials/customize_runtime.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ optimizer = dict(type='Adam', lr=0.0003, weight_decay=0.0001)
 ```
 
 为了修改模型的学习率，使用者仅需要修改配置文件里 optimizer 的 `lr` 即可。
-使用者可以参照 PyTorch 的 [API 文档](https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim)
+使用者可以参照 PyTorch 的 [API 文档](https://pytorch.org/docs/en/stable/optim.html?highlight=optim#module-torch.optim)
```
Comment (gaotongxiao) [inline docs/zh_cn/tutorials/customize_runtime.md:15]: This is a link to pytorch.org :)

File: docs/zh_cn/useful_tools.md [hunk 5]
```diff
@@ -184,7 +184,7 @@ python tools/pytorch2torchscript.py \
 
 先决条件
 
-- 按照 [ONNXRuntime in mmcv](https://mmcv.readthedocs.io/en/latest/onnxruntime_op.html) 和 [TensorRT plugin in mmcv](https://github.com/open-mmlab/mmcv/blob/master/docs/tensorrt_plugin.md) ，用 ONNXRuntime 自定义运算 (custom ops) 和 TensorRT 插件安装 `mmcv-full`
+- 按照 [ONNXRuntime in mmcv](https://mmcv.readthedocs.io/en/latest/onnxruntime_op.html) 和 [TensorRT plugin in mmcv](https://github.com/open-mmlab/mmcv/blob/master/docs/en/tensorrt_plugin.md) ，用 ONNXRuntime 自定义运算 (custom ops) 和 TensorRT 插件安装 `mmcv-full`
```
Comment (gaotongxiao) [inline docs/zh_cn/useful_tools.md:187]: Both are deadlinks

File: docs/zh_cn/useful_tools.md [hunk 6]
```diff
@@ -326,7 +326,7 @@ docker run --rm \
 mmseg-serve:latest
 ```
 
-阅读关于推理 (8080), 管理 (8081) 和指标 (8082) APIs 的 [文档](https://github.com/pytorch/serve/blob/072f5d088cce9bb64b2a18af065886c9b01b317b/docs/rest_api.md) 。
+阅读关于推理 (8080), 管理 (8081) 和指标 (8082) APIs 的 [文档](https://github.com/pytorch/serve/blob/072f5d088cce9bb64b2a18af065886c9b01b317b/docs/en/rest_api.md) 。
```
Comment (gaotongxiao) [inline docs/zh_cn/useful_tools.md:329]: pytorch link

**Review by gaotongxiao — CHANGES_REQUESTED** [review]
Should double-check and revert changes for those links to pytorch.org or any other websites.

File: docs/zh_cn/tutorials/customize_models.md (hunk 3 above)
Comment (MengzhangLI) [reply docs/zh_cn/tutorials/customize_models.md:45]: `https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim` works, `https://pytorch.org/docs/en/stable/optim.html?highlight=optim#module-torch.optim` is wrong.

File: docs/zh_cn/tutorials/customize_runtime.md (hunk 4 above)
Comment (MengzhangLI) [reply docs/zh_cn/tutorials/customize_runtime.md:15]: `https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim` works, `https://pytorch.org/docs/en/stable/optim.html?highlight=optim#module-torch.optim` is wrong.

File: docs/zh_cn/train.md (hunk 2 above)
Comment (MengzhangLI) [reply docs/zh_cn/train.md:62]: Both `https://pytorch.org/docs/en/stable/distributed_deprecated.html#launch-utility` and `https://pytorch.org/docs/stable/distributed_deprecated.html#launch-utility` are dead links.

File: .gitignore (hunk 1 above)
Comment (MengzhangLI) [reply .gitignore:67]: Thanks for reminder, `docs/zh_cn/_build/` has been added.

File: docs/zh_cn/train.md (hunk 2 above)
Comment (MengzhangLI) [reply docs/zh_cn/train.md]: I just copied related url from [mmdet](https://github.com/open-mmlab/mmdetection/blob/master/docs/1_exist_data_model.md), it is `https://pytorch.org/docs/stable/distributed.html#launch-utility`.

File: docs/zh_cn/useful_tools.md (hunk 5 above)
Comment (MengzhangLI) [reply docs/zh_cn/useful_tools.md:187]: Waiting for mmcv merging.

**Comment by Junjun2016** [comment]
Hi @MengzhangLI 
Please fix the conflict.

File: docs/zh_cn/useful_tools.md [hunk 7]
```diff
@@ -140,7 +140,7 @@ python tools/deploy_test.py \
 
 ### 导出 TorchScript (试验性)
 
-我们同样提供一个脚本去把模型导出成 [TorchScript](https://pytorch.org/docs/stable/jit.html) 格式。您可以使用 pytorch C++ API [LibTorch](https://pytorch.org/docs/stable/cpp_index.html) 去推理训练好的模型。
+我们同样提供一个脚本去把模型导出成 [TorchScript](https://pytorch.org/docs/en/stable/jit.html) 格式。您可以使用 pytorch C++ API [LibTorch](https://pytorch.org/docs/en/stable/cpp_index.html) 去推理训练好的模型。
```
Comment (zhouzaida) [inline docs/zh_cn/useful_tools.md:143]:
```suggestion
我们同样提供一个脚本去把模型导出成 [TorchScript](https://pytorch.org/docs/stable/jit.html) 格式。您可以使用 pytorch C++ API [LibTorch](https://pytorch.org/docs/en/stable/cpp_index.html) 去推理训练好的模型。
```

File: docs/en/useful_tools.md [hunk 8]
```diff
@@ -142,7 +142,7 @@ TensorRT is only available on configs with `whole mode`.
 
 ### Convert to TorchScript (experimental)
 
-We also provide a script to convert model to [TorchScript](https://pytorch.org/docs/stable/jit.html) format. You can use the pytorch C++ API [LibTorch](https://pytorch.org/docs/stable/cpp_index.html) inference the trained model. The converted model could be visualized by tools like [Netron](https://github.com/lutzroeder/netron). Besides, we also support comparing the output results between PyTorch and TorchScript model.
+We also provide a script to convert model to [TorchScript](https://pytorch.org/docs/en/stable/jit.html) format. You can use the pytorch C++ API [LibTorch](https://pytorch.org/docs/en/stable/cpp_index.html) inference the trained model. The converted model could be visualized by tools like [Netron](https://github.com/lutzroeder/netron). Besides, we also support comparing the output results between PyTorch and TorchScript model.
```
Comment (zhouzaida) [inline docs/en/useful_tools.md:145]:
```suggestion
We also provide a script to convert model to [TorchScript](https://pytorch.org/docs/stable/jit.html) format. You can use the pytorch C++ API [LibTorch](https://pytorch.org/docs/stable/cpp_index.html) inference the trained model. The converted model could be visualized by tools like [Netron](https://github.com/lutzroeder/netron). Besides, we also support comparing the output results between PyTorch and TorchScript model.
```

File: docs/en/useful_tools.md [hunk 9]
```diff
@@ -336,7 +336,7 @@ docker run --rm \
 mmseg-serve:latest
 ```
 
-[Read the docs](https://github.com/pytorch/serve/blob/072f5d088cce9bb64b2a18af065886c9b01b317b/docs/rest_api.md) about the Inference (8080), Management (8081) and Metrics (8082) APIs
+[Read the docs](https://github.com/pytorch/serve/blob/072f5d088cce9bb64b2a18af065886c9b01b317b/docs/en/rest_api.md) about the Inference (8080), Management (8081) and Metrics (8082) APIs
```
Comment (zhouzaida) [inline docs/en/useful_tools.md:339]:
```suggestion
[Read the docs](https://github.com/pytorch/serve/blob/072f5d088cce9bb64b2a18af065886c9b01b317b/docs/rest_api.md) about the Inference (8080), Management (8081) and Metrics (8082) APIs
```

File: docs/zh_cn/tutorials/training_tricks.md [hunk 10]
```diff
@@ -48,7 +48,7 @@ model=dict(
                         1.0865, 1.0955, 1.0865, 1.1529, 1.0507])))
 ```
 
-`class_weight` 将被作为 `weight` 参数，传递给 `CrossEntropyLoss`。详细信息请参照 [PyTorch 文档](https://pytorch.org/docs/stable/nn.html?highlight=crossentropy#torch.nn.CrossEntropyLoss) 。
+`class_weight` 将被作为 `weight` 参数，传递给 `CrossEntropyLoss`。详细信息请参照 [PyTorch 文档](https://pytorch.org/docs//stable/nn.html?highlight=crossentropy#torch.nn.CrossEntropyLoss) 。
```
Comment (zhouzaida) [inline docs/zh_cn/tutorials/training_tricks.md:51]:
```suggestion
`class_weight` 将被作为 `weight` 参数，传递给 `CrossEntropyLoss`。详细信息请参照 [PyTorch 文档](https://pytorch.org/docs/stable/nn.html?highlight=crossentropy#torch.nn.CrossEntropyLoss) 。
```

File: docs/en/tutorials/training_tricks.md [hunk 11]
```diff
@@ -49,7 +49,7 @@ model=dict(
                         1.0865, 1.0955, 1.0865, 1.1529, 1.0507])))
 ```
 
-`class_weight` will be passed into `CrossEntropyLoss` as `weight` argument. Please refer to [PyTorch Doc](https://pytorch.org/docs/stable/nn.html?highlight=crossentropy#torch.nn.CrossEntropyLoss) for details.
+`class_weight` will be passed into `CrossEntropyLoss` as `weight` argument. Please refer to [PyTorch Doc](https://pytorch.org/docs//stable/nn.html?highlight=crossentropy#torch.nn.CrossEntropyLoss) for details.
```
Comment (zhouzaida) [inline docs/en/tutorials/training_tricks.md:52]:
```suggestion
`class_weight` will be passed into `CrossEntropyLoss` as `weight` argument. Please refer to [PyTorch Doc](https://pytorch.org/docs/stable/nn.html?highlight=crossentropy#torch.nn.CrossEntropyLoss) for details.
```

File: docs/en/tutorials/customize_runtime.md [hunk 12]
```diff
@@ -11,7 +11,7 @@ For example, if you want to use `ADAM` (note that the performance could drop a l
 optimizer = dict(type='Adam', lr=0.0003, weight_decay=0.0001)
 ```
 
-To modify the learning rate of the model, the users only need to modify the `lr` in the config of optimizer. The users can directly set arguments following the [API doc](https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim) of PyTorch.
+To modify the learning rate of the model, the users only need to modify the `lr` in the config of optimizer. The users can directly set arguments following the [API doc](https://pytorch.org/docs//stable/optim.html?highlight=optim#module-torch.optim) of PyTorch.
```
Comment (zhouzaida) [inline docs/en/tutorials/customize_runtime.md:14]:
```suggestion
To modify the learning rate of the model, the users only need to modify the `lr` in the config of optimizer. The users can directly set arguments following the [API doc](https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim) of PyTorch.
```

File: docs/en/tutorials/customize_models.md [hunk 13]
```diff
@@ -44,7 +44,7 @@ For example, if you want to use `ADAM`, though the performance will drop a lot,
 optimizer = dict(type='Adam', lr=0.0003, weight_decay=0.0001)
 ```
 
-The users can directly set arguments following the [API doc](https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim) of PyTorch.
+The users can directly set arguments following the [API doc](https://pytorch.org/docs//stable/optim.html?highlight=optim#module-torch.optim) of PyTorch.
```
Comment (zhouzaida) [inline docs/en/tutorials/customize_models.md:47]:
```suggestion
The users can directly set arguments following the [API doc](https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim) of PyTorch.
```

**Review by zhouzaida — APPROVED** [review]

**Review by gaotongxiao — APPROVED** [review]

**Review by Junjun2016 — APPROVED** [review]
